mm/2s2h/GameInteractor/GameInteractor_VanillaBehavior.h

```diff
@@ -0,0 +1,1953 @@
+    // ((DOORWARP1_GET_FF(&this->dyna.actor) == ENDOORWARP1_FF_4) && CHECK_QUEST_ITEM(QUEST_REMAINS_GYORG)) ||
+    // ((DOORWARP1_GET_FF(&this->dyna.actor) == ENDOORWARP1_FF_5) && CHECK_QUEST_ITEM(QUEST_REMAINS_TWINMOLD))
+    // ```
+    // #### `args`
+    // - `*DoorWarp1`
+    VB_ACTIVATE_BOSS_WARP_PAD,
+
+    // #### `result`
+    // ```c
+    // GET_CUR_UPG_VALUE(UPG_BOMB_BAG) == 3
+    // ```
+    // #### `args`
+    // - `*EnAkindonuts` (unused)
+    VB_AKINDONUTS_CONSIDER_BOMB_BAG_PURCHASED,
+
+    // #### `result`
+    // ```c
+    // !((u32)INV_CONTENT(ITEM_MAGIC_BEANS) != ITEM_MAGIC_BEANS)
+    // ```
+    // #### `args`
+    // - `*EnAkindonuts` (unused)
+    VB_AKINDONUTS_CONSIDER_ELIGIBLE_FOR_BEAN_REFILL,
+
+    // #### `result`
+    // ```c
+    // !(GET_CUR_UPG_VALUE(UPG_BOMB_BAG) < 2)
+    // ```
+    // #### `args`
+    // - `*EnAkindonuts` (unused)
+    VB_AKINDONUTS_CONSIDER_ELIGIBLE_FOR_BOMB_BAG,
+
+    // #### `result`
+    // ```c
+    // Inventory_HasEmptyBottle()
+    // ```
+    // #### `args`
+    // - `*EnAkindonuts`
+    VB_AKINDONUTS_CONSIDER_ELIGIBLE_FOR_POTION_REFILL,
+
+    // #### `result`
+    // ```c
+    // (CURRENT_DAY != 3) || (gSaveContext.save.isNight == 0)
+    // ```
+    // #### `args`
+    // - None
+    VB_ALLOW_SONG_DOUBLE_TIME_ON_FINAL_NIGHT,
+
+    // #### `result`
+    // ```c
+    // EnBal_CheckIfMapUnlocked(this, play)
+    // ```
+    // #### `args`
+    // - `*EnBal` (unused)
+    VB_ALREADY_HAVE_TINGLE_MAP,
+
+    // #### `result`
+    // ```c
+    // true
+    // ```
+    // #### `args`
+    // - `*f32` (speed)
+    VB_APPLY_AIR_CONTROL,
+
+    // #### `result`
+    // ```c
+    // !play->interfaceCtx.perfectLettersOn
+    // ```
+    // #### `args`
+    // - `*EnSyatekiMan` (unused)
+    // - `*s32` (sBonusTimer)
+    VB_ARCHERY_ADD_BONUS_POINTS,
+
+    // #### `result`
+    // ```c
+    // true
+    // ```
+    // #### `args`
+    // - None
+    VB_BALLAD_PLAYED_FORM,
+
+    // #### `result`
+    // ```c
+    // true
+    // ```
+    // #### `args`
+    // - `*EnGinkoMan`
+    VB_BANKER_GIVE_REWARD,
+
+    // #### `result`
+    // ```c
+    // true
+    // ```
+    // #### `args`
+    // - `*Actor`
+    VB_BARREL_OR_CRATE_DROP_COLLECTIBLE,
+
+    // #### `result`
+    // ```c
+    // false
+    // ```
+    // #### `args`
+    // - None
+    VB_BE_CLIMBABLE_SURFACE,
+
+    // #### `result`
+    // ```c
+    // CHECK_WEEKEVENTREG(WEEKEVENTREG_73_80) && !CHECK_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK)
+    // ```
+    // #### `args`
+    // - `*EnBomBowlMan` (unused)
+    VB_BE_ELIGIBLE_FOR_BOMBERS_NOTEBOOK,
+
+    // #### `result`
+    // ```c
+    // SurfaceType_GetData(colCtx, poly, bgId, 1) >> 17 & 1
+    // ```
+    // #### `args`
+    // - `*CollisionPoly` (unused)
+    // - `s32` (bgId, unused)
+    VB_BE_HOOKSHOT_SURFACE,
+
+    // #### `result`
+    // ```c
+    // fabsf(playerPosRelToDoor.z) < 50.0f
+    // ```
+    // #### `args`
+    // - `*f32`
+    VB_BE_NEAR_DOOR,
```

Comment:
While what it's doing is basically the same between the two cases, the variables are just a teensy bit different (straight f32 vs resolving position.z):

```suggestion
    // #### `result`
    // #### In `EnDoor_Idle`, for `*EnDoor`:
    // ```c
    // fabsf(playerPosRelToDoor.z) < 50.0f
    // ```
    // #### `args`
    // - `*f32`
    // #### In `func_808A0E28`, for `*DoorShutter`:
    // ```c
    // fabsf(temp_f0) < 50.0f
    // ```
    // #### `args`
    // - `*f32`
    VB_BE_NEAR_DOOR,
```

Reply:
Good catch.